### PR TITLE
Caching schema between requests

### PR DESCRIPTION
**Pull request by jonnyshaw89**
### What is the context of this PR?
Possible implementation of caching of survey json after the 1st request.

### How to review 
Feedback on approach please.

### CODE REVIEW

**Review by benjefferies — CHANGES_REQUESTED** [review]
Have we ran all the tests against the application in EQ_DEV_MODE=False?

**Comment by jonnyshaw89** [comment]
@benjefferies I have run all the functional tests with caching enabled.

File: app/__init__.py
Comment (benjefferies) [inline app/__init__.py]: cache.init_app(application) will default to the non-caching config

File: tests/integration/test_schema_cache.py
Comment (benjefferies) [inline tests/integration/test_schema_cache.py]: `cache.cache._cache['app.utilities.schema.load_and_parse_schema_memver']` could be changed to use the public method cache.get('app.utilities.schema.load_and_parse_schema_memver')

File: tests/integration/test_schema_cache.py
Comment (benjefferies) [inline tests/integration/test_schema_cache.py]: Would we be able to apply this to all of the integration tests to run closer to production?

**Review by benjefferies — CHANGES_REQUESTED** [review]

**Review by benjefferies — APPROVED** [review]

File: app/__init__.py
Comment (benjefferies) [inline app/__init__.py:130]: 👍

File: tests/integration/test_schema_cache.py
Comment (iwootten) [inline tests/integration/test_schema_cache.py:11]: Hmm, this is a bit weird. Can we import from app.cache or 'as' another name to save drilling down this far?

File: tests/integration/test_schema_cache_off.py
Comment (iwootten) [inline tests/integration/test_schema_cache_off.py]: and again here?

**Review by iwootten — CHANGES_REQUESTED** [review]

File: tests/integration/test_schema_cache.py
Comment (jonnyshaw89) [reply tests/integration/test_schema_cache.py:11]: Im not sure what you mean? can you give an example?

Comment (iwootten) [reply tests/integration/test_schema_cache.py:11]: from app.cache import cache
or
    
    from app.cache import cache as app_cache

Comment (jonnyshaw89) [reply tests/integration/test_schema_cache.py:11]: ah ok. but if i did that then i would still need to higher level import for the next assertion on the cache on line 20 `cache.get`

Comment (iwootten) [reply tests/integration/test_schema_cache.py:11]: Oh, didn't spy that, fair enough.

**Review by iwootten — APPROVED** [review]

**Comment by benjefferies** [comment]
Are there any outstanding questions to be answered or can this be merged?

**Comment by jonnyshaw89** [comment]
The only open question it how much of a performance gain we get. but we can't run that at the moment but i don't see a reason that should block merge?

**Comment by benjefferies** [comment]
Discussed with @ajmaddaford and it's probably worth running a performance test locally before merging. This will save the overhead of rolling back if it's not better

**Comment by jonnyshaw89** [comment]
In progress

**Comment by jonnyshaw89** [comment]
Results from test with 100 users.
Average survey completion time
Before 693ms
After 607ms

**Pull request merged**
